app/lib/three_scale/api/collection.rb

```diff
@@ -36,7 +38,21 @@ def respond_to_missing?(method_name, include_private = false)
 
   private
 
-  def pagination
-    pagination_attrs(@collection)
+  attr_reader :collection
+
+  def represented
+    @represented ||= collection.try(:represented) || collection
+  end
+
+  def pagination_attrs(wrapper: nil)
+    return {} if represented.per_page >= represented.total_entries
+
+    pagination = %i[per_page total_entries total_pages current_page].each_with_object({}) do |meta_field, pagination_hash|
```

Comment:
We could extract these into a constant for more visibility:
```suggestion
    pagination = METADATA.each_with_object({}) do |meta_field, pagination_hash|
```